### PR TITLE
Pgcraft misc

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: tests/texture.py [hunk 1]
```diff
@@ -425,6 +425,8 @@ def texture_2d_array_mipmap(cfg: ngl.SceneCfg):
 void main()
 {
     float x = floor(var_uvcoord.x * steps) / steps;
+    float t = step(0.5, var_uvcoord.y);
+    x = mix(x, 1.0 - x, t);
```
Comment (ubitux) [inline tests/texture.py]: can you explain what you are doing here? I tried to graph what's going on here, and it looks like a slope rotating non linearly around (½,½), but that's not exactly helping.

Comment (mbouron) [reply tests/texture.py]: I just want the original color x for the upper part of the texture [0 -> 0.5] and its inverse for the lower part [0.5, 1].

File: doc/usr/expl/shaders.md [hunk 2]
```diff
@@ -161,10 +161,10 @@ To sample a `Texture2D` from within the shader, multiple options are available:
 Other variables related to textures are exposed to different stages of the shader:
 
 Availability | Type   | Name              | Description
--------------|-------|-------------------|------------
-`Texture*`   |`mat4` | `%s_coord_matrix` | uv transformation matrix of the texture associated with the render node using key `%s`, it should be applied to the geometry uv coordinates `ngl_uvcoord` to obtain the final texture coordinates. This variable is always exposed in the `vertex` stage.
-`Texture2D`  |`vec2` | `%s_dimensions`   | dimensions in pixels of the texture associated with the render node using key `%s`. This variable is exposed in the same stage as the texture.
-`Texture2D`  |`float`| `%s_ts`           | timestamp generated by the texture data source, 0.0f for images and buffers, frame timestamp for audios and videos. This variable is exposed in the same stage as the texture.
+-------------|--------|-------------------|------------
```
[the remark this answers is not in the document]
Comment (mbouron) [reply doc/usr/expl/shaders.md]: Removed this chunk.

File: tests/texture.py (hunk 1 above)
Comment (ubitux) [reply tests/texture.py]: ah ok yeah so `x = var_uvcoord.y < 0.5 ? x : 1 - x` 😅

Comment (mbouron) [reply tests/texture.py]: yup.

Comment (mbouron) [reply tests/texture.py]: Adjusted to this more readable form.

File: tests/texture.py [hunk 3]
```diff
@@ -538,6 +539,8 @@ def texture_3d(cfg: ngl.SceneCfg):
...
```
Comment (ubitux) [inline tests/texture.py:543]: you forgot to adjust this one

**Review by ubitux — CHANGES_REQUESTED** [review]

File: tests/texture.py (hunk 3 above)
Comment (mbouron) [reply tests/texture.py]: My bad, fixed.

**Review by ubitux — APPROVED** [review]

**Pull request merged**